jni/src/main/java/org/infinispan/client/hotrod/impl/RemoteCacheImpl.java

```diff
@@ -626,5 +632,34 @@ public ServerStatistics stats() {
         return stats;
     }
 
+    @Override
+    public <T> T execute(String scriptName, Map<String, ?> params) {
+        MapType argsMap = new MapType();
+        JBossMarshaller jb = new JBossMarshaller();
+        for (String parName : params.keySet()) {
+            ByteBuffer b;
+            try {
+                b= jb.objectToBuffer(params.get(parName));
+                argsMap.set(parName, new String(b.getBuf()));
+            } catch (IOException | InterruptedException e) {
+                // TODO Auto-generated catch block
+                e.printStackTrace();
+            }
+        }
+        UCharVector executeResult = jniRemoteCache.execute(scriptName, argsMap);
+        byte[] buf = new byte[(int)executeResult.size()];
+        for (int i=0; i< buf.length; i++)
+        {
+            buf[i]= (byte)executeResult.get(i);
+        }
+        try {
+            return (T) jb.objectFromByteBuffer(buf);
+        } catch (ClassNotFoundException | IOException e) {
+            // TODO Auto-generated catch block
```

Comment:
Remove?

Reply:
the exceptions need to be caught to respect the method signature. Re throwing as runtime exception

Reply:
I was referring to removing the `TODO` comment

Reply:
yeah, I've realized too late :D